hyperapp.d.ts

```diff
@@ -54,42 +59,48 @@ export type ActionResult<State extends Hyperapp.State> =
   | null
   | void
 
-export type ActionCallers<
+export interface Action<State extends Hyperapp.State> {
+  (data: any): ActionResult<State>
+}
+
+export interface Actions<State extends Hyperapp.State> {
+  // should be Actions<State[action]>, but this is not allowed by typescript
+  [action: string]: Actions<any> | Action<State>
+}
+
+export interface InternalAction<
```

Comment:
Is this a `ModuleAction`?

Reply:
This is the type that represents the implementation of the action (i.e. `(state, actions, data) = ...`.
I liked the name `InternalAction` that Swizz suggested, but it seems everyone prefers `ActionCaller` (even though it it not something that calls an action, it's the code that implements the actions).

Reply:
Is there a link to the JavaScript documentation for this? I'm not sure where I would use this right now.

Reply:
Sorry I don't understand your comment: documentation for actions?

Reply:
I meant this https://github.com/hyperapp/hyperapp/blob/master/docs/actions.md

Reply:
I am sorry but I still don't understand what you want me to change now? Could you explicitly say what is wrong with the code at this point?

Thanks!